sdk/eventhub/azure-eventhub/azure/eventhub/_client_base.py

```diff
@@ -202,6 +202,9 @@ def __init__(self, fully_qualified_namespace, eventhub_name, credential, **kwarg
         self._container_id = CONTAINER_PREFIX + str(uuid.uuid4())[:8]
         if isinstance(credential, AzureSasCredential):
             self._credential = AzureSasTokenCredential(credential)
+        elif isinstance(credential, AzureNamedKeyCredential):
+            name, key = credential.named_key
+            self._credential = EventHubSharedKeyCredential(name, key) # type: ignore
```

Comment:
if users rotate name and key in AzureNamedKeyCredential, then `EventHubSharedKeyCredential` would not be able to reflect the change?